libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -238,6 +239,7 @@ void ContractCompiler::appendBaseConstructor(FunctionDefinition const& _construc
 		solAssert(m_baseArguments.count(&_constructor), "");
 		std::vector<ASTPointer<Expression>> const* arguments = m_baseArguments[&_constructor];
 		solAssert(arguments, "");
+		solAssert(arguments->size() == constructorType.parameterTypes().size(), "");
```

Comment:
Do we have a test case for this? I think it may silently accept less parameters at the moment.

Reply:
It fails properly, but we actually don't have a test for it.

Reply:
Added.